tools/topology/topology2/include/pipelines/cavs/host-gateway-capture.conf

```diff
@@ -105,6 +105,43 @@ Class.Pipeline."host-gateway-capture" {
 				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
 				out_ch_map	$CHANNEL_MAP_3_POINT_1
 			}
+			# 32-bit 48KHz 1ch
+			Object.Base.audio_format.7 {
+				in_channels		1
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				out_channels		1
+				in_ch_cfg	$CHANNEL_CONFIG_MONO
+				in_ch_map	$CHANNEL_MAP_MONO
+				out_ch_cfg	$CHANNEL_CONFIG_MONO
+				out_ch_map	$CHANNEL_MAP_MONO
+			}
+			# 24-bit 48KHz 1ch
+			Object.Base.audio_format.8 {
+				in_channels		1
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	24
+				out_channels		1
+				in_ch_cfg	$CHANNEL_CONFIG_MONO
+				in_ch_map	$CHANNEL_MAP_MONO
+				out_ch_cfg	$CHANNEL_CONFIG_MONO
+				out_ch_map	$CHANNEL_MAP_MONO
+			}
+			# 16-bit 48KHz 1ch
+			Object.Base.audio_format.9 {
+				in_channels		1
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		1
+				in_ch_cfg	$CHANNEL_CONFIG_MONO
+				in_ch_map	$CHANNEL_MAP_MONO
+				out_ch_cfg	$CHANNEL_CONFIG_MONO
+				out_ch_map	$CHANNEL_MAP_MONO
+			}
```

Comment:
This is a generic change that impacts all solutions. Does this really work with e.g. HDaudio and DMIC?

I don't follow the logic where we would have to propagate all restrictions from a 3rd party codec all the way to the host-gateway and create a giant union of all requirements.

Reply:
agree that this can't go in the class definition. Will need to go in the specific instantiation for the pipeline that needs this

Reply:
@plbossart @ranj063 But this is about extending the format capacity. The 1 ch format will be selected only if we do 1 channel capture. I.e `arecord -c 1`. Nothing will be different if we do `arecord -c 2`.

Reply:
some applications such as PulseAudio will try all possible hw_params.

if you expose support for 1ch, and it does not work on a specific platform, there's an issue.

We only want to expose 1ch for the platforms where it actually works.

Reply:
> some applications such as PulseAudio will try all possible hw_params.
> 
> if you expose support for 1ch, and it does not work on a specific platform, there's an issue.
> 
> We only want to expose 1ch for the platforms where it actually works.

@plbossart That is exactly the second commit does. We will set SDW_JACK_CAPTURE_CH = 1 in Cmakefile if the codec support 1 channel capture only. SDW_JACK_CAPTURE_CH is 2 by default which means the existing topologies support 2 channel format only. In other words, the supported formats doesn't change after the 1st commit extending the available formats. It is more like to add 1 channel format support in NHLT.

Reply:
@udaymb @RDharageswari @yongzhi1

Reply:
> The first commit changes host-gateway-capture.conf, so it's not a SoundWire-only change at all @bardliao
> 
> This is changing EVERY host DMA.

Yes, but only one of the formats in host-gateway-capture.conf will be selected and the 1 channel formats will be selected only when we do 1 channel capture. And this will not happen unless we set channels_min/channels_max to 1 in PCM caps.

Reply:
Ack, so as long as the pcm_caps is not change to by default expose 1-2 channel range to ALSA, this will have limited effect to other scenarios.

Reply:
> agree that this can't go in the class definition. Will need to go in the specific instantiation for the pipeline that needs this

@ranj063 Sorry, I didn't get your point. Adding available formats in the class definition can benefit all pipelines that use the class. I didn't get why we need to add available formats in specific pipelines.

Reply:
@bardliao but the point is not all pipeline need to support 1ch capture